Nu/Nu/Ecs/EcsPrelude.fs

```diff
@@ -45,32 +47,75 @@ type Store =
         end
 
 /// Stores components for an Ecs.
-type Store<'c when 'c : struct and 'c :> 'c Component> (name) =
+type Store<'c when 'c: struct and 'c :> 'c Component>(name) =
     let mutable arr = Array.zeroCreate<'c> Constants.Ecs.ArrayReserve
     member this.Length = arr.Length
     member this.Name = name
     member this.Item i = &arr.[i]
     member this.SetItem index comp = arr.[index] <- comp
     member this.ZeroItem index = arr.[index] <- Unchecked.defaultof<'c>
-    member this.Grow () =
+
+    member this.Grow() =
         let length = int (single (max arr.Length 2) * 1.5f)
         let arr' = Array.zeroCreate<'c> length
-        Array.Copy (arr, arr', arr.Length)
+        Array.Copy(arr, arr', arr.Length)
         arr <- arr'
-    member this.Read index count (stream : FileStream) =
+
+    /// <summary>
+    /// Writes entities to the current stream
+    /// </summary>
+    /// <exception cref="IndexOutOfRangeException"><paramref name="index"/> is out of range.</exception>
+    /// <param name="index">The offset in internal collection at which to begin storing the data read from the current stream.</param>
+    /// <param name="count">The number of entities to be read from the current stream.</param>
+    /// <param name="stream">Input stream to read data from</param>
+    member this.Write index count (stream: FileStream) =
```

Comment:
The whole problem with this approach is that we can't guarantee the safety of it. 'c is a managed struct, and managed structs cannot by blit as bytes reliably. We really would have to stick with (presumably) slower facilities like marshalling.